compiler/runtime/Runtime.hpp

```diff
@@ -170,31 +170,69 @@ enum TR_RuntimeHelper
 class TR_RuntimeHelperTable
    {
 public:
-   void* getAddress(TR_RuntimeHelper h)
-      {
-      return h < TR_numRuntimeHelpers ? _helpers[h] : (void *) (uintptr_t) 0xDEADB00F;
-      }
+
+   /**
+    * \brief
+    *
+    * For helper functions, this API returns the raw entry address. If the runtime
+    * helper's corresponding linkage is not a function, return the helper as data.
+    *
+    */
+   void* getAddress(TR_RuntimeHelper h);
+
+   /**
+    * \brief
+    *
+    * Looks up the given helper index and returns the environment pointer on applicable platforms.
+    * Certain platforms have function descriptors for each C function. And the environment pointer
+    * is a field in the function descriptor.
+    */
+   void* getEnvironment(TR_RuntimeHelper h);
```

Comment:
I fully support @ymanton's review as well. These all sound like excellent improvements!

Reply:
There's another source of confusion I want to mention: the array in `TR_RuntimeHelperTable` contains a mixture of entry points and constant values. How about naming them `getFunctionPointer` and `getEntryPointOrConst`.

1. `getFunctionPointer` returns function pointers for helper functions. It shall return `#define INVALID_FUNC_POINTER 0xdeadbeaf` if the index refers to a constant value. This allows API users (CHelperLinkage on z, for instance) to get function descriptors and extract whatever fields in there.
2. `getFunctionEntryPointOrConst` will replace the current `getAddress()` and return unwrapped entry points or constant numbers. The rest of the codebase assumes it can query raw entry point or constant via the same API.

Also, I want to give `TR_RuntimeHelperTable` a constructor to initialize the whole linkage array to TR_None. This class lacks initialization.

Reply:
That's fine. I'm a little surprised to hear that constants are being kept in there, but if that's the case then yes that makes sense.

Reply:
I was too. I don't think they belong there and we should eventually weed them out. Until then I agree we should make the API explicit about the kinds of things we can (currently) extract.

Reply:
Grep for `SET_CONST` and you'll see a dozen of helpers are just constants.